crates/net/network/src/state.rs

```diff
@@ -480,6 +480,8 @@ impl<N: NetworkPrimitives> NetworkState<N> {
                 self.on_session_closed(peer)
             }
 
+            let has_received_responses = !received_responses.is_empty();
+
             for (peer_id, resp) in received_responses {
                 if let Some(action) = self.on_eth_response(peer_id, resp) {
                     self.queued_messages.push_back(action);
```

Comment:
if I look at `on_eth_response` then the Option this returns is actually guaranteed to be `None`, right?
so we could remove this logic entirely and make this cheaper by removing the optional return type?

Reply:
Updated in Updated in cc691ba